### PR TITLE
Revise wording for Secret concept

### PR DESCRIPTION
**Pull request by sftim**
This change is a subset of the changes from PR #24169 (a PR that has proved too big to review).

I will follow this up with further PRs to incorporate the other changes that #24169 suggests.

/sig storage

### CODE REVIEW

File: content/en/docs/concepts/configuration/secret.md [hunk 1]
```diff
@@ -12,17 +12,21 @@ weight: 30
...
-{{< glossary_tooltip term_id="pod" >}} definition or in a
-{{< glossary_tooltip text="container image" term_id="image" >}}.
-See [Secrets design document](https://git.k8s.io/community/contributors/design-proposals/auth/secrets.md) for more information.
-
 A Secret is an object that contains a small amount of sensitive data such as
 a password, a token, or a key. Such information might otherwise be put in a
-Pod specification or in an image. Users can create Secrets and the system
-also creates some Secrets.
+{{< glossary_tooltip term_id="pod" >}} specification or in a 
+{{< glossary_tooltip text="container image" term_id="image" >}}; using a
```
Comment (shannonxtreme) [inline content/en/docs/concepts/configuration/secret.md:18]: {{< glossary_tooltip text="container image" term_id="image" >}}. Using a...

File: content/en/docs/concepts/configuration/secret.md [hunk 2]
```diff
@@ -12,17 +12,21 @@ weight: 30
...
+Secret means that you don't need to include confidential data in your
+application code.
+
+Because Secrets can be created independently of the Pods that use them, there
+is less risk of the Secret (and its data) being exposed during the workflow of
+creating, viewing, and editing Pods. Kubernetes, and applications that run in
+your cluster, can also take additional precautions with Secrets, such as avoiding
+writing secret data to nonvolatile storage.
```
Comment (shannonxtreme) [inline content/en/docs/concepts/configuration/secret.md:26]: avoid using "secret" to describe the data inside a Secret. Say sensitive or confidential instead.

File: content/en/docs/concepts/configuration/secret.md [hunk 3]
```diff
@@ -31,7 +35,10 @@ access, or anyone with access to Kubernetes' underlying data store, etcd. In
 order to safely use Secrets, it is recommended you (at a minimum):
 
 1. [Enable Encryption at Rest](/docs/tasks/administer-cluster/encrypt-data/) for Secrets.
-2. [Enable or configure RBAC rules](/docs/reference/access-authn-authz/authorization/) that restrict reading and writing the Secret. Be aware that secrets can be obtained implicitly by anyone with the permission to create a Pod.
+2. [Enable or configure RBAC rules](/docs/reference/access-authn-authz/authorization/) that
+   restrict reading and writing the Secret. Be aware that secrets can be obtained
```
Comment (shannonxtreme) [inline content/en/docs/concepts/configuration/secret.md:39]: restrict reading from and writing to the Secret.

File: content/en/docs/concepts/configuration/secret.md [hunk 4]
```diff
@@ -31,7 +35,10 @@ access, or anyone with access to Kubernetes' underlying data store, etcd. In
...
+   implicitly by anyone with the permission to create a Pod.
```
Comment (shannonxtreme) [inline content/en/docs/concepts/configuration/secret.md:40]: 1. Be aware that **Secrets** can be obtained.... (capitalize S)
2. What does "can be obtained implicitly" mean? Maybe reword to "Be aware that anyone with the permission to create a pod can retrieve Secrets."?

File: content/en/docs/concepts/configuration/secret.md [hunk 5]
```diff
@@ -47,6 +54,10 @@ A Secret can be used with a Pod in three ways:
 - As [container environment variable](#using-secrets-as-environment-variables).
 - By the [kubelet when pulling images](#using-imagepullsecrets) for the Pod.
 
+The Kubernetes control plane also uses Secrets; for example,
```
Comment (shannonxtreme) [inline content/en/docs/concepts/configuration/secret.md:57]: also uses Secrets. For example, ...

**Review by shannonxtreme — CHANGES_REQUESTED** [review]
Added some suggestions for clarity and consistency :)

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-27716--kubernetes-io-master-staging.netlify.app/docs/concepts/configuration/secret/

File: content/en/docs/concepts/configuration/secret.md [hunk 6]
```diff
@@ -12,17 +12,21 @@ weight: 30
...
 <!-- overview -->
 
-Kubernetes Secrets let you store and manage sensitive information, such
-as passwords, OAuth tokens, and ssh keys. Storing confidential information in a Secret
-is safer and more flexible than putting it verbatim in a
-{{< glossary_tooltip term_id="pod" >}} definition or in a
-{{< glossary_tooltip text="container image" term_id="image" >}}.
-See [Secrets design document](https://git.k8s.io/community/contributors/design-proposals/auth/secrets.md) for more information.
-
 A Secret is an object that contains a small amount of sensitive data such as
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/secret.md:15]: I think it would help to add that a Secret is a core Kubernetes object (or something similar to that text)?
https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/secret-v1/#get-read-the-specified-secret

Comment (sftim) [reply content/en/docs/concepts/configuration/secret.md:15]: This change is a subset of the changes from PR #24169 (a PR that has proved too big to review).

I will follow this up with further PRs to incorporate the other changes that #24169 suggests. Anyone else can make those follow-ups; it doesn't have to be me.

File: content/en/docs/concepts/configuration/secret.md (hunk 4 above)
Comment (sftim) [reply content/en/docs/concepts/configuration/secret.md:40]: This change is a subset of the changes from PR #24169 (a PR that has proved too big to review).

I will follow this up with further PRs to incorporate the other changes that #24169 suggests. Anyone else can make those follow-ups; it doesn't have to be me.

File: content/en/docs/concepts/configuration/secret.md (hunk 3 above)
Comment (sftim) [reply content/en/docs/concepts/configuration/secret.md:39]: This change is a subset of the changes from PR #24169 (a PR that has proved too big to review).

I will follow this up with further PRs to incorporate the other changes that #24169 suggests. Anyone else can make those follow-ups; it doesn't have to be me.

File: content/en/docs/concepts/configuration/secret.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/concepts/configuration/secret.md:26]: Good idea!

File: content/en/docs/concepts/configuration/secret.md (hunk 6 above)
Comment (kbhawkey) [reply content/en/docs/concepts/configuration/secret.md:15]: Yes, that's nice.

File: content/en/docs/concepts/configuration/secret.md [hunk 7]
```diff
@@ -12,26 +12,33 @@ weight: 30
...
+
+Secrets are similar to {{< glossary_tooltip text="ConfigMaps" term_id="configmap" >}}
+but are specifically intended to hold confidential data.
 
 {{< caution >}}
-Kubernetes Secrets are, by default, stored as unencrypted base64-encoded
-strings. By default they can be retrieved - as plain text - by anyone with API
-access, or anyone with access to Kubernetes' underlying data store, etcd. In
-order to safely use Secrets, it is recommended you (at a minimum):
+Kubernetes Secrets are, by default, stored unencrypted in the API server's underlying data store (etcd). Anyone with API access can retrieve or modify a Secret, and so can anyone with access to etcd.
```
Comment (kbhawkey) [inline content/en/docs/concepts/configuration/secret.md:32]: @sftim, These changes look good.
Should these new lines be wrapped at around the same character length as the other lines?

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-27716--kubernetes-io-master-staging.netlify.app/docs/concepts/configuration/secret/

**Comment by sftim** [comment]
Rebased, should be ready for review.

**Comment by celestehorgan** [comment]
I'm making an executive call here, @sftim, and choosing to treat the remaining unresolved issues as non-blocking :) 

Thank you so much for your hard work on this. 

/lgtm

**Comment by zhangguanzhang** [comment]
@tengqm need sync to zh

**Comment by tengqm** [comment]
@zhangguanzhang With dev-1.22 "becoming" the new "main" branch, we have a lot of pages to resync. Will initiate an sprint to get this done. Thanks for reminding.
